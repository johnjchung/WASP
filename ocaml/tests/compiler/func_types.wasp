Func float float_func() {
    return 1.0;
}

Func int int_func() {
    return 1;
}

Func string string_func() {
    return "hey";
}

Func bool bool_func() {
    return true;
}

Endpoint ftoc(GET:) {
    float a = float_func();
    int b = int_func();
    string c = string_func();
    bool d = bool_func();
}
