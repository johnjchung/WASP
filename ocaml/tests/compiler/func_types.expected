package main

import (
	"github.com/gin-gonic/gin"
	"math"
	"os"
)

func int_conv(anything interface{}) int {
	switch v := anything.(type) {
		case int:
			return v
		case float64, float32:
			var float_var = v.(float64)
			return int(math.Trunc(float_var))
		default:
			return 0
	}
	return 0
}

func float_conv(anything interface{}) float64 {
 	 switch v := anything.(type) {
		case int:
            return float64(v)
		case float64:
			return v
		default:
			return 0
	}
	return 0.0
}

type ftoc_struct struct {
	A float64 `db:"a" json:"a"`
	B int `db:"b" json:"b"`
	C string `db:"c" json:"c"`
	D bool `db:"d" json:"d"`
}

func main() {
	router := gin.Default()
	router.GET("/ftoc", Get_ftoc)
	router.Run(":"+os.Getenv("PORT"))
}

func float_func() float64 {
	return 1.
}

func int_func() int {
	return 1
}

func string_func() string {
	return "hey"
}

func bool_func() bool {
	return true
}

func ftoc_struct_wrapper() ftoc_struct {
	return ftoc_struct{A: float_func(), B: int_func(), C: string_func(), D: bool_func()}
}

func Get_ftoc(c *gin.Context) {
	ftoc_struct := ftoc_struct_wrapper()
	c.JSON(200, ftoc_struct)
}
